aria-practices.html

```diff
@@ -984,75 +975,55 @@ <h5>Tree Popup Keyboard Interaction</h5>
-              <li>When the combobox popup is visible, the combobox element contains or owns an element that has role <a href="#listbox" class="role-reference">listbox</a>, <a href="#tree" class="role-reference">tree</a>, <a href="#grid" class="role-reference">grid</a>, or <a href="#dialog" class="role-reference">dialog</a>.</li>
-              <li>
-                If the combobox popup has a role other than <code>listbox</code>, the element with role <code>combobox</code> has <a href="#aria-haspopup" class="property-reference">aria-haspopup</a> set to a value that corresponds to the popup type.
-                That is, <code>aria-haspopup</code> is set to <code>grid</code>, <code>tree</code>, or <code>dialog</code>.
-                Note that elements with role <code>combobox</code> have an implicit <code>aria-haspopup</code> value of <code>listbox</code>.
-              </li>
-              <li>When the combobox popup is visible, the textbox element has <a href="#aria-controls" class="property-reference">aria-controls</a> set to a value that refers to the combobox popup element.</li>
-            </ul>
-          </li>
-          <li>In a combobox implementing the ARIA 1.0 pattern:
-            <ul>
-              <li>The element that serves as the textbox has role <a href="#combobox" class="role-reference">combobox</a>.</li>
-              <li>When the combobox popup is visible, the element with role combobox has <a href="#aria-owns" class="property-reference">aria-owns</a> set to a value that refers to an element with role <a href="#listbox" class="role-reference">listbox</a>.</li>
-              <li>
-                the element with role <code>combobox</code> has a value for <a href="#aria-haspopup" class="property-reference">aria-haspopup</a> of <code>listbox</code>.
-                Note that elements with role <code>combobox</code> have an implicit <code>aria-haspopup</code> value of <code>listbox</code>.
-              </li>
-            </ul>
+          <li>The popup is an element that has role <a href="#listbox" class="role-reference">listbox</a>, <a href="#tree" class="role-reference">tree</a>, <a href="#grid" class="role-reference">grid</a>, or <a href="#dialog" class="role-reference">dialog</a>.</li>
+          <li>
+            If the popup has a role other than <code>listbox</code>, the element with role <code>combobox</code> has <a href="#aria-haspopup" class="property-reference">aria-haspopup</a> set to a value that corresponds to the popup type.
+            That is, <code>aria-haspopup</code> is set to <code>grid</code>, <code>tree</code>, or <code>dialog</code>.
+            Note that elements with role <code>combobox</code> have an implicit <code>aria-haspopup</code> value of <code>listbox</code>.
           </li>
+          <li>When the popup is visible, the combobox element has <a href="#aria-controls" class="property-reference">aria-controls</a> set to a value that refers to the popup element.</li>
```

Comment:
Should we remove "When the popup is visible"? If I understand correctly, the popup element is always present in the DOM, and `aria-controls` should be present always. The popup is just styled with `display: none` when hidden.

Reply:
Changed in https://github.com/w3c/aria-practices/pull/1255/commits/e6e8499495b3084fa3d131c88ebca414e0164eda

Reply:
The actual requirement is aria-controls only needs to be present when the popup is visible. There is discussion about what browsers should expose when aria-controls refers to something not present or not visible. That said, it is not an error to refer to an element that is not visible.

Reply:
I revised this text to make the actual requirement more clear but specify that it is valid to refer to an element that is not visible.